<a href="https://www.kaggle.com/code/divyamuthukrishnan/study-on-type-of-dwellings-population-canada?scriptVersionId=102382068" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/housing-affordability-in-canada/Structural-dwellings-household-size.csv
/kaggle/input/housing-affordability-in-canada/HPI 1981-2022 by regions.csv
/kaggle/input/housing-affordability-in-canada/income-distribution-2012-2020.csv
/kaggle/input/housing-affordability-in-canada/HPI 1981-2022 federal and provincial.csv
/kaggle/input/housing-affordability-in-canada/population_dwellings_count.csv
/kaggle/input/housing-affordability-in-canada/Interest and mortgage rates 1951-2022.csv
/kaggle/input/housing-affordability-in-canada/population-by-region-1946-2022.csv
/kaggle/input/housing-affordability-in-canada/CPI-inflation-by-region-1914-202.csv
/kaggle/input/housing-affordability-in-canada/housing-supply-price-rental.csv
/kaggle/input/housing-affordability-in-canada/housing-supply-and-rental/housing-supply-and-rental/ottawa_section1_.csv
/kaggle/input/housing-affordability-in-canada/housing-supply-and-rental/housing-supply-and-rental/saguenay_section1_.csv
/kaggle/input/housing-aff

In [2]:
import pandas as pd


In [3]:
pop_df = pd.read_csv('../input/housing-affordability-in-canada/population-by-region-1946-2022.csv')
pop_df.head()

REF_DATE     GEO           DGUID VECTOR  COORDINATE  Population estimate
0   Jan-46  Canada  2016A000011124     v1           1             12188000
1   Apr-46  Canada  2016A000011124     v1           1             12241000
2   Jul-46  Canada  2016A000011124     v1           1             12316000
3   Oct-46  Canada  2016A000011124     v1           1             12393000
4   Jan-47  Canada  2016A000011124     v1           1             12450000

# Descriptive Statistics

In [4]:
pop_df.describe()

COORDINATE  Population estimate
count  3839.000000         3.839000e+03
mean      7.257880         3.989061e+06
std       4.013029         7.278025e+06
min       1.000000         9.000000e+03
25%       4.000000         1.517775e+05
50%       7.000000         9.412970e+05
75%      11.000000         4.196062e+06
max      15.000000         3.865474e+07

In [5]:
pop_df.head()

REF_DATE     GEO           DGUID VECTOR  COORDINATE  Population estimate
0   Jan-46  Canada  2016A000011124     v1           1             12188000
1   Apr-46  Canada  2016A000011124     v1           1             12241000
2   Jul-46  Canada  2016A000011124     v1           1             12316000
3   Oct-46  Canada  2016A000011124     v1           1             12393000
4   Jan-47  Canada  2016A000011124     v1           1             12450000

# Data Visualization

## 1. EDA on Population growth 

In [6]:
import plotly.express as px

In [7]:
fig = px.line(pop_df, x="REF_DATE", y="Population estimate", color = 'GEO', title='Line chart for Population growth in Canada')
fig.show()

In [8]:
fig = px.scatter(pop_df, x="REF_DATE", y="Population estimate", color="GEO", title = 'Scatter plot for Population growth in Canada')
fig.show()

From the above scatterplot we can interpret the following:
* Population of Canada keeps on increasing from 1946 to 2022
* Ontario has the highest population all the time and that increases steadily.
* Next comes Qubec and then British Columbia and then Alberta.
* After 1970, the growth of population in Alberta and British Columbia increased.
* All the other places have significantly small number of population and is not increasing that much.

In [9]:
fig = px.bar(pop_df, x="REF_DATE",  y="Population estimate", color="GEO")
fig.show()

In [10]:
pop_df.head()

REF_DATE     GEO           DGUID VECTOR  COORDINATE  Population estimate
0   Jan-46  Canada  2016A000011124     v1           1             12188000
1   Apr-46  Canada  2016A000011124     v1           1             12241000
2   Jul-46  Canada  2016A000011124     v1           1             12316000
3   Oct-46  Canada  2016A000011124     v1           1             12393000
4   Jan-47  Canada  2016A000011124     v1           1             12450000

In [11]:
# pop_df[pop_df['REF_DATE'].str.extract('^[pP].*')>0]
pop_yearly_df = pop_df.copy()
pop_yearly_df = pop_yearly_df[pop_yearly_df['REF_DATE'].str.match('^Jan.*')== True]

In [12]:
pop_yearly_df.head()

REF_DATE     GEO           DGUID VECTOR  COORDINATE  Population estimate
0    Jan-46  Canada  2016A000011124     v1           1             12188000
4    Jan-47  Canada  2016A000011124     v1           1             12450000
8    Jan-48  Canada  2016A000011124     v1           1             12710000
12   Jan-49  Canada  2016A000011124     v1           1             12998000
16   Jan-50  Canada  2016A000011124     v1           1             13607000

In [13]:

pop_yearly_df = pop_yearly_df[pop_yearly_df.GEO != 'Canada']

In [14]:
pop_yearly_df.head()

REF_DATE                        GEO        DGUID VECTOR  COORDINATE  \
49   Jan-52  Newfoundland and Labrador  2016A000210     v2           2   
50   Jan-52       Prince Edward Island  2016A000211     v8           3   
51   Jan-52                Nova Scotia  2016A000212     v9           4   
52   Jan-52              New Brunswick  2016A000213    v10           5   
53   Jan-52                     Quebec  2016A000224    v11           6   

    Population estimate  
49               368000  
50               100000  
51               647000  
52               521000  
53              4126000

In [15]:
fig = px.bar(pop_yearly_df, x='REF_DATE', y='Population estimate',
             color='GEO',
             labels={'Population estimate':'Population'}, height=400)           #

fig.show()


In [16]:
df = pop_df.query("REF_DATE == 'Jan-22'") #.query("continent == 'Europe'")
df = df[df.GEO != 'Canada']
fig = px.pie(df, values='Population estimate', names='GEO', title='Population of Canada by Province in Jan 2022')
fig.show()

**The pie chart above shows the percentage of population in all Provinces and territorries as on Jan 2022**

# 2. EDA on Dwellings count and structural type dwelling

In [17]:
df = pd.read_csv('../input/housing-affordability-in-canada/population_dwellings_count.csv')

In [18]:
df.head()

REF_DATE                        GEO           DGUID  Coordinate  \
0      2021                     Canada  2021A000011124           1   
1      2021  Newfoundland and Labrador     2021A000210           2   
2      2021                Bay Roberts   2021S05100046           3   
3      2021             Bishop's Falls   2021S05100072           4   
4      2021                  Bonavista   2021S05100082           5   

   Population,2021  Population,2016  Population percentage change  \
0         36991981         35151728                           5.2   
1           510550           519716                          -1.8   
2             6897             6896                           0.0   
3             3006             3087                          -2.6   
4             2979             3140                          -5.1   

   Total private dwellings,2021  \
0                      16284235   
1                        269184   
2                          3184   
3                          1498   
4                          1628   

   Private dwellings occupied by usual residents, 2021  \
0                                           14978941     
1                                             223253     
2                                               2888     
3                                               1386     
4                                               1379     

   Land area in square kilometres,2021  \
0                           8788702.80   
1                            358170.37   
2                                16.88   
3                                 6.89   
4                                 4.36   

   Population density per square kilometre,2021  
0                                           4.2  
1                                           1.4  
2                                         408.5  
3                                         436.1  
4                                         683.1

In [19]:
str_dwell_df = pd.read_csv('../input/housing-affordability-in-canada/Structural-dwellings-household-size.csv')

In [20]:
str_dwell_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1566 entries, 0 to 1565
Data columns (total 8 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   REF_DATE                                 1566 non-null   int64  
 1   GEO                                      1566 non-null   object 
 2   DGUID                                    1566 non-null   object 
 3   Structural type of dwelling              1566 non-null   object 
 4   Coordinate                               1566 non-null   float64
 5   Total - Household                        1566 non-null   int64  
 6   Number of persons in private households  1566 non-null   int64  
 7   Average household size                   1556 non-null   float64
dtypes: float64(2), int64(3), object(3)
memory usage: 98.0+ KB


In [21]:
str_dwell_df.head()

REF_DATE     GEO           DGUID          Structural type of dwelling  \
0      2021  Canada  2021A000011124  Total - Structural type of dwelling   
1      2021  Canada  2021A000011124                Single-detached house   
2      2021  Canada  2021A000011124                  Semi-detached house   
3      2021  Canada  2021A000011124                            Row house   
4      2021  Canada  2021A000011124        Apartment or flat in a duplex   

   Coordinate  Total - Household  Number of persons in private households  \
0         1.1           14978940                                 36328475   
1         1.2            7872310                                 21496250   
2         1.3             746560                                  2004855   
3         1.4             980105                                  2539830   
4         1.5             821490                                  2012905   

   Average household size  
0                     2.4  
1                     2.7  
2                     2.7  
3                     2.6  
4                     2.5

There are 10 missing values in the 'Average household size' column that I missed. I'm going to drop those rows.

train.describe(include=['object']).T.style    



In [22]:
rows_with_nan = str_dwell_df[str_dwell_df['Average household size'].isna()]
rows_with_nan

REF_DATE                                   GEO           DGUID  \
24        2021               Corner Brook (CA), N.L.    2021S0504015   
178       2021  Campbellton (Quebec part) (CA), Que.  2021S050624330   
276       2021         Dolbeau-Mistassini (CA), Que.    2021S0504411   
420       2021   Salaberry-de-Valleyfield (CA), Que.    2021S0504465   
665       2021                  Ingersoll (CA), Ont.    2021S0504533   
863       2021                  Stratford (CA), Ont.    2021S0504553   
906       2021               Wasaga Beach (CA), Ont.    2021S0504565   
1177      2021        Lloydminster (CA), Alta./Sask.    2021S0504840   
1455      2021                 Salmon Arm (CA), B.C.    2021S0504920   
1518      2021              Williams Lake (CA), B.C.    2021S0504950   

                            Structural type of dwelling  Coordinate  \
24    Apartment in a building that has five or more ...         3.7   
178                         Other single-attached house        20.8   
276   Apartment in a building that has five or more ...        31.7   
420   Apartment in a building that has five or more ...        47.7   
665                                    Movable dwelling        74.9   
863                                    Movable dwelling        96.9   
906   Apartment in a building that has five or more ...       101.7   
1177                        Other single-attached house       131.8   
1455  Apartment in a building that has five or more ...       162.7   
1518  Apartment in a building that has five or more ...       169.7   

      Total - Household  Number of persons in private households  \
24                    5                                        5   
178                   5                                        5   
276                   5                                        5   
420                   5                                        0   
665                   5                                        5   
863                   5                                        5   
906                   5                                        5   
1177                  0                                        5   
1455                  5                                       10   
1518                  5                                        5   

      Average household size  
24                       NaN  
178                      NaN  
276                      NaN  
420                      NaN  
665                      NaN  
863                      NaN  
906                      NaN  
1177                     NaN  
1455                     NaN  
1518                     NaN

In [23]:
str_dwell_df = str_dwell_df.dropna()     #Dropped all the rows with NaN

In [24]:
str_dwell_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1556 entries, 0 to 1565
Data columns (total 8 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   REF_DATE                                 1556 non-null   int64  
 1   GEO                                      1556 non-null   object 
 2   DGUID                                    1556 non-null   object 
 3   Structural type of dwelling              1556 non-null   object 
 4   Coordinate                               1556 non-null   float64
 5   Total - Household                        1556 non-null   int64  
 6   Number of persons in private households  1556 non-null   int64  
 7   Average household size                   1556 non-null   float64
dtypes: float64(2), int64(3), object(3)
memory usage: 109.4+ KB


In [25]:
str_dwell_df.describe()

REF_DATE   Coordinate  Total - Household  \
count    1556.0  1556.000000       1.556000e+03   
mean     2021.0    88.024679       5.537476e+04   
std         0.0    50.208331       4.929918e+05   
min      2021.0     1.100000       0.000000e+00   
25%      2021.0    44.575000       2.500000e+02   
50%      2021.0    88.000000       1.270000e+03   
75%      2021.0   131.425000       8.372500e+03   
max      2021.0   174.900000       1.497894e+07   

       Number of persons in private households  Average household size  
count                             1.556000e+03             1556.000000  
mean                              1.345758e+05                2.048586  
std                               1.224228e+06                0.564586  
min                               0.000000e+00                0.000000  
25%                               5.050000e+02                1.800000  
50%                               2.482500e+03                2.100000  
75%                               1.915375e+04                2.400000  
max                               3.632848e+07                4.500000

In [26]:
str_dwell_df.describe(include=['object']).T.style    


In [27]:
fig = px.bar(str_dwell_df, x="Structural type of dwelling",  y="Total - Household", color = 'GEO')
fig.show()

In [28]:

df = str_dwell_df[str_dwell_df['Structural type of dwelling'] != 'Total - Structural type of dwelling']

fig = px.pie(df, values='Total - Household', names='Structural type of dwelling', title='Structural type of dwelling - 2022')
fig.show()

From the above figure we can interpret the following:
* 50% of dwelling is Single deteched house
* The next most common type of dwelling is Apartment in a building that has fewer than five storeys

In [29]:
str_dwell_df.head()

REF_DATE     GEO           DGUID          Structural type of dwelling  \
0      2021  Canada  2021A000011124  Total - Structural type of dwelling   
1      2021  Canada  2021A000011124                Single-detached house   
2      2021  Canada  2021A000011124                  Semi-detached house   
3      2021  Canada  2021A000011124                            Row house   
4      2021  Canada  2021A000011124        Apartment or flat in a duplex   

   Coordinate  Total - Household  Number of persons in private households  \
0         1.1           14978940                                 36328475   
1         1.2            7872310                                 21496250   
2         1.3             746560                                  2004855   
3         1.4             980105                                  2539830   
4         1.5             821490                                  2012905   

   Average household size  
0                     2.4  
1                     2.7  
2                     2.7  
3                     2.6  
4                     2.5

In [30]:

pd.set_option('display.max_rows', None)
str_dwell_df['GEO'].value_counts()


Canada                                          9
Weyburn (CA), Sask.                             9
Saskatchewan                                    9
Estevan (CA), Sask.                             9
Moose Jaw (CA), Sask.                           9
North Battleford (CA), Sask.                    9
Prince Albert (CA), Sask.                       9
Regina (CMA), Sask.                             9
Saskatoon (CMA), Sask.                          9
Swift Current (CA), Sask.                       9
Yorkton (CA), Sask.                             9
Ottawa - Gatineau (Ontario part) (CMA), Ont.    9
Alberta                                         9
Brooks (CA), Alta.                              9
Calgary (CMA), Alta.                            9
Camrose (CA), Alta.                             9
Canmore (CA), Alta.                             9
Edmonton (CMA), Alta.                           9
Grande Prairie (CA), Alta.                      9
High River (CA), Alta.                          9


In [31]:
# list of Province and territories in Canada

prov_terr = ['Alberta', 'British Columbia', 'Manitoba', 'New Brunswick', 'Nova Scotia','Ontario', 'Prince Edward Island', 'Quebec', 'Saskatchewan', 'Newfoundland and Labrador',
                'Northwest Territories','Nunavut','Yukon']


In [32]:
str_dwell_prov_df = str_dwell_df[str_dwell_df['GEO'].isin(prov_terr)]

In [33]:
str_dwell_prov_df.GEO.value_counts()

Newfoundland and Labrador    9
Prince Edward Island         9
Nova Scotia                  9
New Brunswick                9
Quebec                       9
Ontario                      9
Manitoba                     9
Saskatchewan                 9
Alberta                      9
British Columbia             9
Yukon                        9
Northwest Territories        9
Nunavut                      9
Name: GEO, dtype: int64

In [34]:
str_dwell_prov_df.head(9)

REF_DATE                        GEO        DGUID  \
9       2021  Newfoundland and Labrador  2021A000210   
10      2021  Newfoundland and Labrador  2021A000210   
11      2021  Newfoundland and Labrador  2021A000210   
12      2021  Newfoundland and Labrador  2021A000210   
13      2021  Newfoundland and Labrador  2021A000210   
14      2021  Newfoundland and Labrador  2021A000210   
15      2021  Newfoundland and Labrador  2021A000210   
16      2021  Newfoundland and Labrador  2021A000210   
17      2021  Newfoundland and Labrador  2021A000210   

                          Structural type of dwelling  Coordinate  \
9                 Total - Structural type of dwelling         2.1   
10                              Single-detached house         2.2   
11                                Semi-detached house         2.3   
12                                          Row house         2.4   
13                      Apartment or flat in a duplex         2.5   
14  Apartment in a building that has fewer than fi...         2.6   
15  Apartment in a building that has five or more ...         2.7   
16                        Other single-attached house         2.8   
17                                   Movable dwelling         2.9   

    Total - Household  Number of persons in private households  \
9              223250                                   502100   
10             161410                                   381190   
11               8695                                    18675   
12              10770                                    21785   
13              27535                                    57690   
14              12330                                    18285   
15                810                                     1060   
16                410                                      815   
17               1295                                     2600   

    Average household size  
9                      2.3  
10                     2.4  
11                     2.2  
12                     2.0  
13                     2.1  
14                     1.5  
15                     1.3  
16                     2.0  
17                     2.0

In [35]:
fig = px.bar(str_dwell_prov_df, x="Structural type of dwelling", y="Total - Household",
             color='GEO', barmode='group',height=700)
fig.show()

In [36]:
fig = px.bar(str_dwell_prov_df, x="GEO", y="Total - Household",
             color='Structural type of dwelling', barmode='group')
fig.show()

In [37]:

fig = px.bar(str_dwell_prov_df[str_dwell_prov_df.GEO == 'Ontario'], x="GEO", y="Total - Household",
             color='Structural type of dwelling', barmode='group')
fig.show()

In [38]:
fig = px.bar(str_dwell_prov_df, x="Structural type of dwelling", y="Total - Household",
             color='GEO')
fig.show()

In [39]:
str_dwell_prov_df.head()

REF_DATE                        GEO        DGUID  \
9       2021  Newfoundland and Labrador  2021A000210   
10      2021  Newfoundland and Labrador  2021A000210   
11      2021  Newfoundland and Labrador  2021A000210   
12      2021  Newfoundland and Labrador  2021A000210   
13      2021  Newfoundland and Labrador  2021A000210   

            Structural type of dwelling  Coordinate  Total - Household  \
9   Total - Structural type of dwelling         2.1             223250   
10                Single-detached house         2.2             161410   
11                  Semi-detached house         2.3               8695   
12                            Row house         2.4              10770   
13        Apartment or flat in a duplex         2.5              27535   

    Number of persons in private households  Average household size  
9                                    502100                     2.3  
10                                   381190                     2.4  
11                                    18675                     2.2  
12                                    21785                     2.0  
13                                    57690                     2.1

In [40]:
# fig = px.bar(str_dwell_prov_df, x="Structural type of dwelling", y="Total - Household", facet_row="GEO")

fig = px.bar(str_dwell_prov_df, x="GEO", y="Total - Household", facet_col="Structural type of dwelling", facet_col_wrap=2, height = 1500)
fig.show()

# 3. EDA on CPI 

In [41]:
CPI_df = pd.read_csv('../input/housing-affordability-in-canada/CPI-inflation-by-region-1914-202.csv')
CPI_df.head()

REF_DATE     GEO           DGUID Products and product groups       UOM  \
0      1914  Canada  2016A000011124                   All-items  2002=100   
1      1914  Canada  2016A000011124        All-items (1992=100)  1992=100   
2      1914  Canada  2016A000011124          Goods and services  2002=100   
3      1915  Canada  2016A000011124                   All-items  2002=100   
4      1915  Canada  2016A000011124        All-items (1992=100)  1992=100   

   UOM_ID     VECTOR  COORDINATE  CPI  
0      17  v41693271       2.200  6.0  
1       7  v41713433       2.309  7.2  
2      17  v41693519       2.273  6.0  
3      17  v41693271       2.200  6.1  
4       7  v41713433       2.309  7.3

In [42]:
CPI_all_df = CPI_df[CPI_df['Products and product groups'] == 'All-items']
CPI_all_df.head()

REF_DATE     GEO           DGUID Products and product groups       UOM  \
0       1914  Canada  2016A000011124                   All-items  2002=100   
3       1915  Canada  2016A000011124                   All-items  2002=100   
6       1916  Canada  2016A000011124                   All-items  2002=100   
9       1917  Canada  2016A000011124                   All-items  2002=100   
12      1918  Canada  2016A000011124                   All-items  2002=100   

    UOM_ID     VECTOR  COORDINATE  CPI  
0       17  v41693271         2.2  6.0  
3       17  v41693271         2.2  6.1  
6       17  v41693271         2.2  6.7  
9       17  v41693271         2.2  7.9  
12      17  v41693271         2.2  8.9

In [43]:
CPI_all_df.tail()

REF_DATE                                 GEO             DGUID  \
85998      2021         Vancouver, British Columbia      2011S0503933   
86003      2021          Victoria, British Columbia      2011S0503935   
86008      2021                   Whitehorse, Yukon  2011A00056001009   
86132      2021  Yellowknife, Northwest Territories  2011A00056106023   
86256      2021                    Iqaluit, Nunavut  2011A00056204003   

      Products and product groups       UOM  UOM_ID     VECTOR  COORDINATE  \
85998                   All-items  2002=100      17  v41695228        27.2   
86003                   All-items  2002=100      17  v41695234        28.2   
86008                   All-items  2002=100      17  v41694896        29.2   
86132                   All-items  2002=100      17  v41695020        30.2   
86256                   All-items  2002=100      17  v41713462        31.2   

         CPI  
85998  138.5  
86003  133.6  
86008  138.9  
86132  142.0  
86256  133.2

In [44]:
# Return a Series containing counts of unique values in column UOM
CPI_all_df.UOM.value_counts()

2002=100    1434
Name: UOM, dtype: int64

Checked how many unique values are in UOM. Usually when comparing the CPI we have to use the same reference year.
Here only one unique value is present and the reference year is 2002 = 100, So we can use CPI data from 1914 - 2021 for comparing.

In [45]:
fig = px.line(CPI_all_df, x="REF_DATE", y="CPI", color = 'GEO', title='Line chart for CPI in Canada')
fig.show()

After 1971 only provincial data is available. So plotting the CPI for year(1971 - 2021)

In [46]:
fig = px.line( CPI_all_df[ (CPI_all_df.REF_DATE > 1970) & ( CPI_all_df['GEO'].isin(prov_terr) )], x="REF_DATE", y="CPI", color = 'GEO', title='Line chart for CPI in Canadian Provinces and territorries(1971-2021)')
fig.show()

In [47]:
fig = px.line( CPI_all_df[ (CPI_all_df.REF_DATE > 1970) & ( CPI_all_df['GEO']=='Canada' )], x="REF_DATE", y="CPI", color = 'GEO', title='Line chart for CPI in Canada(1971-2021)')
fig.show()

In [48]:
fig = px.bar( CPI_all_df[ (CPI_all_df.REF_DATE > 1970) & ( CPI_all_df['GEO'] =='Canada' )], x="REF_DATE", y="CPI", color = 'GEO', title='Bar chart for CPI in Canada(1971-2021)')
fig.show()

In [49]:
fig = px.bar( CPI_all_df[ (CPI_all_df.REF_DATE > 2010) & ( CPI_all_df['GEO'].isin(prov_terr) )], x="REF_DATE", y="CPI", color = 'GEO', title='CPI in Canada(2010-2021)',barmode = 'group')
fig.show()

In [50]:
fig = px.bar( CPI_df[ (CPI_df.REF_DATE == 2021) & ( CPI_df['GEO']=='Canada' )], x="Products and product groups", y="CPI", title='CPI for different products in Canada 2021', height = 1000)
fig.show()

### Top inflated products in Canada in 2021

In [51]:
CPI_df[ (CPI_df.REF_DATE == 2021) & ( CPI_df['GEO']=='Canada' ) & ( CPI_df.CPI > 200 ) ].sort_values('CPI', ascending=False)

REF_DATE     GEO           DGUID  \
84395      2021  Canada  2016A000011124   
84564      2021  Canada  2016A000011124   
84400      2021  Canada  2016A000011124   
84562      2021  Canada  2016A000011124   
84563      2021  Canada  2016A000011124   
84402      2021  Canada  2016A000011124   
84541      2021  Canada  2016A000011124   
84467      2021  Canada  2016A000011124   
84408      2021  Canada  2016A000011124   
84549      2021  Canada  2016A000011124   
84487      2021  Canada  2016A000011124   
84289      2021  Canada  2016A000011124   
84453      2021  Canada  2016A000011124   

                             Products and product groups       UOM  UOM_ID  \
84395            Homeowners' home and mortgage insurance  2002=100      17   
84564       Other tobacco products and smokers' supplies  2002=100      17   
84400                                              Water  2002=100      17   
84562             Tobacco products and smokers' supplies  2002=100      17   
84563                                         Cigarettes  2002=100      17   
84402                           Fuel oil and other fuels  2002=100      17   
84541              Video and audio subscription services  2002=100      17   
84467                                          Jewellery  2002=100      17   
84408           Postal and other communications services  2002=100      17   
84549                                         Newspapers  2002=100      17   
84487                                       Parking fees  2002=100      17   
84289                               Fresh or frozen beef  2002=100      17   
84453  Services related to household furnishings and ...  2002=100      17   

          VECTOR  COORDINATE    CPI  
84395  v41693357       2.880  275.3  
84564  v41693516       2.268  269.4  
84400  v41693362       2.930  266.2  
84562  v41693514       2.266  264.8  
84563  v41693515       2.267  264.6  
84402  v41693364       2.950  243.6  
84541  v41693493       2.245  225.6  
84467  v41693419       2.169  220.3  
84408  v41693369       2.100  214.4  
84549  v41693501       2.253  209.4  
84487  v41693442       2.192  206.9  
84289  v41693276       2.700  202.4  
84453  v41693405       2.138  201.0

* CPI is highest for 'Home owners home and mortgage insurance'(275.3)
* then comes 'Other tobacco products and smokers' supplies', 'Water,'Tobacco products and smokers' supplies', Cigarettes
* After that 'Fuel oil and other fuels'

In [52]:
CPI_df['UOM'].value_counts()

2002=100      83921
1992=100       1328
200212=100      494
2013=100        419
201812=100       45
201104=100       30
200704=100       14
2017=100          6
Name: UOM, dtype: int64

In [53]:
CPI_df[ CPI_df['UOM'] == '2013=100' ].head()

REF_DATE     GEO           DGUID        Products and product groups  \
67718      2013  Canada  2016A000011124                   Bacon (2013=100)   
67719      2013  Canada  2016A000011124  Ham excluding deli ham (2013=100)   
67725      2013  Canada  2016A000011124           Canned salmon (2013=100)   
67726      2013  Canada  2016A000011124             Canned tuna (2013=100)   
67728      2013  Canada  2016A000011124      Shrimps and prawns (2013=100)   

            UOM  UOM_ID      VECTOR  COORDINATE    CPI  
67718  2013=100      25  v122586782       2.316  100.0  
67719  2013=100      25  v122586783       2.317  100.0  
67725  2013=100      25  v122586784       2.318  100.0  
67726  2013=100      25  v122586785       2.319  100.0  
67728  2013=100      25  v122586786       2.320  100.0

References:
1. https://plotly.com/python/basic-charts/
2. https://plotly.com/python/facet-plots/
3. https://www.bls.gov/cpi/factsheets/cpi-math-calculations.pdf